libfwupdplugin/fu-smbios.c

```diff
@@ -89,6 +93,76 @@ fu_smbios_convert_dt_string (FuSmbios *self, guint8 type, guint8 offset,
 	fu_smbios_convert_dt_value (self, type, offset, item->strings->len);
 }
 
+#if defined (__FreeBSD__) || defined (__DragonFly__)
+static void
+fu_smbios_kenv_sysctl_string (FuSmbios *self, guint8 type, guint8 offset,
+				const gchar *buf, gsize bufsz)
+{
+	FuSmbiosItem *item = g_ptr_array_index (self->items, type);
+
+	/* add to strtab */
+	g_ptr_array_add (item->strings, g_strndup (buf, bufsz));
+	fu_smbios_convert_dt_value (self, type, offset, item->strings->len);
+}
+
+static void
+fu_smbios_kenv_lookup (const char* sminfo, gchar* buf)
+{
+	g_debug ("DMI request for %s", sminfo);
+	if (kenv (KENV_GET, sminfo, buf, sizeof (buf) - 1) == -1)
+		g_debug ("Cannot get %s", sminfo);
+	else
+		g_debug ("%s: %s", sminfo, buf);
+}
+
+static void
+fu_smbios_convert_kenv_string (FuSmbios *self, guint8 type, guint8 offset,
+				const char* sminfo)
+{
+	/* Maximum value length - 128 */
+	gchar buf[128] = {};
+	gsize bufsz = sizeof(buf);
+
+	fu_smbios_kenv_lookup (sminfo, buf);
+	fu_smbios_kenv_sysctl_string (self, type, offset, buf, bufsz);
```

Comment:
surely you don't want to call this if fu_smbios_kenv_lookup doesn't set a value?

Reply:
Fixed.